PostHog/PostHogQueue.swift

```diff
@@ -75,41 +73,51 @@ class PostHogQueue {
         }
     }
 
-    func start() {
-        // Setup the monitoring of network status for the queue
-        reachability?.whenReachable = { reachability in
-            self.pausedLock.withLock {
-                if self.config.dataMode == .wifi, reachability.connection != .wifi {
-                    hedgeLog("Queue is paused because its not in WiFi mode")
-                    self.paused = true
-                } else {
-                    self.paused = false
+    func start(disableReachabilityForTesting: Bool,
+               disableQueueTimerForTesting: Bool)
+    {
+        if !disableReachabilityForTesting {
```

Comment:
Those flags are always enabled but during tests the `reachability` and `timer` just give us a hard time during debugging, also because the reachability goes crazy, we'd need a deboucer just for testing.